packages/cli/src/util/runner.ts

```diff
@@ -0,0 +1,43 @@
+import { getHttpOperationsFromResource } from '@stoplight/prism-http';
+import { IPrismHttpServer } from '@stoplight/prism-http-server/src/types';
+import * as chokidar from 'chokidar';
+import { CreatePrismOptions } from './createServer';
+
+type CreatePrism = (options: CreatePrismOptions) => Promise<IPrismHttpServer | undefined>;
+
+export function runPrismAndSetupWatcher(createPrism: CreatePrism, options: CreatePrismOptions, spec: string) {
+  return createPrism(options).then(server => {
+    if (server) {
+      const watcher = chokidar.watch(spec, { awaitWriteFinish: { stabilityThreshold: 500, pollInterval: 100 } });
+
+      watcher.on('change', () => {
+        server!.fastify.log.info('Restarting Prism...');
```

Comment:
This is because the reference could change by the time this even triggers. You can avoid this by using another variable and write

`const s = server;` and use this one